File: packages/fern-java/fern-api-model/src/main/java/com/errors/ErrorDefinition.java
package com.errors;

import com.StagedBuilderStyle;
import com.commons.interfaces.IWithDocs;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.types.NamedType;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableErrorDefinition.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ErrorDefinition extends IWithDocs {
  NamedType name();

  List<ErrorProperty> properties();

  Optional<HttpErrorConfiguration> http();

  static ImmutableErrorDefinition.NameBuildStage builder() {
    return ImmutableErrorDefinition.builder();
  }
}

File: packages/compiler/ir-types/fern-ir-typescript/src/model/errors/ErrorDefinition.ts
import { WithDocs } from "../commons/WithDocs";
import { NamedType } from "../types/NamedType";
import { Type } from "../types/Type";
import { HttpErrorConfiguration } from "./HttpErrorConfiguration";

export interface ErrorDefinition extends WithDocs {
    name: NamedType;
    type: Type;
    http: HttpErrorConfiguration | null | undefined;
}
